packages/playwright-core/src/server/screenshotter.ts

```diff
@@ -267,8 +267,9 @@ export class Screenshotter {
     if (!process.env.PW_TEST_SCREENSHOT_NO_FONTS_READY) {
       progress.log('waiting for fonts to load...');
       await Promise.all(this._page.frames().map(async frame => {
-        await frame.nonStallingEvaluateInExistingContext('document.fonts.ready', false, 'utility').catch(() => {});
-      }));
+        const utilityContext = await frame._utilityContext();
```

Comment:
This is opposite of what we agreed on though, wouldn't you want to go for the non-stalling evaluate? `nonStallingRawEvaluateInExistingMainContext` in this case.

Reply:
Discussed offline to only wait for fonts on the frame of the locator/mainframe.